packages/core/admin/server/services/permission/permissions-manager/sanitize.js

```diff
@@ -108,8 +108,12 @@ module.exports = ({ action, ability, model }) => {
   const omitCreatorRoles = omit([`${CREATED_BY_ATTRIBUTE}.roles`, `${UPDATED_BY_ATTRIBUTE}.roles`]);
 
   const pickAllowedAdminUserFields = ({ attribute, key, value }, { set }) => {
-    if (attribute.type === 'relation' && attribute.target === 'admin::user') {
-      set(key, pick(['id', 'firstname', 'lastname', 'username'], value));
+    if (attribute.type === 'relation' && attribute.target === 'admin::user' && value) {
+      if (Array.isArray(value)) {
+        set(key, value.map(val => pick(['id', 'firstname', 'lastname', 'username'], val)), value);
+      } else {
+        set(key, pick(['id', 'firstname', 'lastname', 'username'], value));
+      }
```

Comment:
WDYT? (here we can use `pick` directly in the `map` since it has a fixed arity of 2 in `lodash/fp`)

```suggestion
    const pickAllowedFields = pick(['id', 'firstname', 'lastname', 'username']);

    if (attribute.type === 'relation' && attribute.target === 'admin::user' && value) {
      if (Array.isArray(value)) {
        set(key, value.map(pickAllowedFields));
      } else {
        set(key, pickAllowedFields(value));
      }
    }
```

Reply:
Looks good to me! I'll make a commit :)